pkg/controller/docdb/dbcluster/dbcluster.go

```diff
@@ -0,0 +1,304 @@
+	obj.EngineVersion = cr.Spec.ForProvider.EngineVersion
+	obj.MasterUserPassword = cr.Spec.ForProvider.MasterUserPassword
+	obj.Port = cr.Spec.ForProvider.Port
+	obj.PreferredBackupWindow = cr.Spec.ForProvider.PreferredBackupWindow
+	obj.PreferredMaintenanceWindow = cr.Spec.ForProvider.PreferredMaintenanceWindow
+	obj.VpcSecurityGroupIds = cr.Spec.ForProvider.VPCSecurityGroupIDs
+
+	// Should apply immediately?
+	obj.ApplyImmediately = awsclient.Bool(true)
+
+	return nil
+}
+
+func (e *hooks) postUpdate(ctx context.Context, cr *svcapitypes.DBCluster, resp *svcsdk.ModifyDBClusterOutput, upd managed.ExternalUpdate, err error) (managed.ExternalUpdate, error) {
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	meta.SetExternalName(cr, awsclient.StringValue(resp.DBCluster.DBClusterIdentifier))
+
+	return upd, nil
+}
+
+func (e *hooks) preCreate(ctx context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.CreateDBClusterInput) error {
+	obj.AvailabilityZones = cr.Spec.ForProvider.AvailabilityZones
+	obj.BackupRetentionPeriod = cr.Spec.ForProvider.BackupRetentionPeriod
+	obj.DBClusterIdentifier = cr.Spec.ForProvider.DBClusterIdentifier
+	obj.DBClusterParameterGroupName = cr.Spec.ForProvider.DBClusterParameterGroupName
+	obj.DBSubnetGroupName = cr.Spec.ForProvider.DBSubnetGroupName
+	obj.DeletionProtection = cr.Spec.ForProvider.DeletionProtection
+	obj.EnableCloudwatchLogsExports = cr.Spec.ForProvider.EnableCloudwatchLogsExports
+	obj.Engine = cr.Spec.ForProvider.Engine
+	obj.EngineVersion = cr.Spec.ForProvider.EngineVersion
+	obj.KmsKeyId = cr.Spec.ForProvider.KMSKeyID
+
+	// TODO: Credentials from secret?
+	obj.MasterUserPassword = cr.Spec.ForProvider.MasterUserPassword
+	obj.MasterUsername = cr.Spec.ForProvider.MasterUsername
+
+	obj.Port = cr.Spec.ForProvider.Port
+	obj.PreSignedUrl = cr.Spec.ForProvider.PreSignedURL
+	obj.PreferredBackupWindow = cr.Spec.ForProvider.PreferredBackupWindow
+	obj.PreferredMaintenanceWindow = cr.Spec.ForProvider.PreferredMaintenanceWindow
+	obj.StorageEncrypted = cr.Spec.ForProvider.StorageEncrypted
+	obj.Tags = generateTags(cr.Spec.ForProvider.Tags)
+	obj.VpcSecurityGroupIds = cr.Spec.ForProvider.VPCSecurityGroupIDs
+
+	return nil
+}
+
+func (e *hooks) postCreate(ctx context.Context, cr *svcapitypes.DBCluster, resp *svcsdk.CreateDBClusterOutput, upd managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	meta.SetExternalName(cr, awsclient.StringValue(resp.DBCluster.DBClusterIdentifier))
+
+	return upd, nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.DeleteDBClusterInput) (bool, error) {
+	// Skip if cluster is already in deleting state
+	if awsclient.StringValue(cr.Status.AtProvider.Status) == svcapitypes.DocDBInstanceStateDeleting {
+		return true, nil
+	}
+
+	obj.DBClusterIdentifier = awsclient.String(meta.GetExternalName(cr))
+	obj.SkipFinalSnapshot = awsclient.Bool(true)
```

Comment:
We need user to configure this. It'd be worth adding a field to `CustomDBClusterParameters` struct for that.

Reply:
Fixed